services/blockstorage/commit_block.go

```diff
@@ -63,11 +61,11 @@ func (s *Service) commitBlock(ctx context.Context, input *services.CommitBlockIn
 	s.metrics.lastCommittedTime.Update(int64(input.BlockPair.TransactionsBlock.Header.Timestamp()))
 
 	if notifyNodeSync {
-		govnr.Once(logfields.GovnrErrorer(logger), func() {
-			shortCtx, cancel := context.WithTimeout(ctx, time.Second) // TODO V1 move timeout to configuration
-			defer cancel()
-			s.nodeSync.HandleBlockCommitted(shortCtx)
-		})
+		select {
```

Comment:
it actually doesn't - the `default` clause prevents us from waiting

Reply:
@gadcl The more "correct" handling from LH perspective would have been to change LH so it will wrap onCommit() with a goroutine, or write to a non-blocking channel.
But, since we are talking about a full rewrite anyway, no point in this additional complication now.